src/api/userApi.ts

```diff
@@ -1,29 +1,44 @@
 import { Auth } from 'aws-amplify';
+import { CognitoUser } from '@aws-amplify/auth';
 import { CognitoUserSession } from 'amazon-cognito-identity-js';
 import { Dispatch } from 'redux';
 import i18n from '../translations/i18n';
-import { USE_MOCK_DATA } from '../utils/constants';
+import { USE_MOCK_DATA, AMPLIFY_REDIRECTED_KEY } from '../utils/constants';
 import { setNotification } from '../redux/actions/notificationActions';
 import { NotificationVariant } from '../types/notification.types';
 
-export const getCurrentUserAttributes = async () => {
-  const loggedIn = localStorage.getItem('amplify-signin-with-hostedUI');
+export const getCurrentUserAttributes = async (retryNumber: number = 0): Promise<any> => {
   try {
-    const cognitoUser = await Auth.currentAuthenticatedUser();
-    const loggedInUser = await cognitoUser?.getSession(async (error: any, session: CognitoUserSession) => {
-      if (error || !session.isValid()) {
-        const currentSession = await Auth.currentSession();
-        cognitoUser.refreshSession(currentSession.getRefreshToken());
-      } else {
-        return (await Auth.currentSession()).getIdToken().payload;
-      }
-    });
-    if (loggedInUser) {
-      return loggedInUser;
+    const currentSession: CognitoUserSession = await Auth.currentSession();
+    const currentSessionData = currentSession.getIdToken().payload;
+
+    if (
+      !currentSession.isValid() ||
+      currentSessionData['custom:cristinId'] === undefined ||
+      currentSessionData['custom:customerId'] === undefined
+    ) {
+      const cognitoUser: CognitoUser = await Auth.currentAuthenticatedUser();
+
+      // Refresh session
+      const refreshedSession: CognitoUserSession = await new Promise((resolve) => {
+        cognitoUser.refreshSession(currentSession.getRefreshToken(), (error, session) => {
+          resolve(session);
+        });
+      });
+      const refreshedSessionData = refreshedSession.getIdToken().payload;
+      return refreshedSessionData;
+    } else {
+      return currentSessionData;
     }
   } catch {
-    if (loggedIn) {
-      return { error: i18n.t('feedback:error.get_user') };
+    // Don't do anything if user is not supposed to be logged in
```

Comment:
```suggestion
    // Don't do anything if user is not supposed to be connected to a customer institution
```
?
Det er vel lov å logge inn uansett?

Reply:
"logged in" er faktisk riktig her. vi vil alltid prøve å hente current user, men hvis brukeren ikke er logget inn blir det en exception. Derfor sjekker vi i localstorage for om brukeren i hele tatt har prøvd å logge inn, slik at vi ikke viser "kunne ikke hente innlogget bruker" om man ikke skal være innlogget uansett

Reply:
ok, men hvordan blir det med en bruker som logger inn som ikke er kunde av NVA nå da? har du testet det?

Reply:
ok, så den kommentaren der er ikke relatert til det som står under da? for hvis du har AMPLIFY_REDIRECTED_KEY så er du jo logget inn?

Reply:
testet med feidebruker som ikke er kunde av NVA nå forresten. Fungerte fint!

Reply:
har ikke prøvd med bruker som ikke er kunde nei, så fint å få det bekreftet!

hvis du har AMPLIFY_REDIRECTED_KEY betyr det at du har prøvd å logge inn, altså har vært på den Feide-innloggingssiden (eller at den bare ikke har blitt fjerna fra gammel innlogging, som vi gjør manuelt i `Login.tsx`). Ting kan fortsatt feile etterpå, både om brukeren har gyldig token eller ikke